crypto/rsa/rsa_x931.c

```diff
@@ -57,6 +57,14 @@ int RSA_padding_check_X931(unsigned char *to, int tlen,
     const unsigned char *p;
 
     p = from;
+    /* Accept even zero-padded input */
+    if (flen == num) {
+        if (*(p++) != 0) {
+            RSAerr(RSA_F_RSA_PADDING_CHECK_SSLV23, RSA_R_BLOCK_TYPE_IS_NOT_02);
+            return -1;
+        }
+        flen--;
+    }
     if ((num != flen) || ((*p != 0x6A) && (*p != 0x6B))) {
```

Comment:
This looks wrong, we expect something from num == flen, so doing flen-- in case flen == num doesn't make sense.

Reply:
Right, this padding scheme already didn't allow leading zeros, so I don't think the new block is needed at all, and the `RSA_R_BLOCK_TYPE_IS_NOT_02` bit seems to be a mistake.